test/mail_helper.js

```diff
@@ -3,101 +3,144 @@
 
 /* eslint-disable no-console */
-var config = require('../config').getProperties()
+
+'use strict'
+const MailParser = require('mailparser').MailParser
+const simplesmtp = require('simplesmtp')
+const P = require('../lib/promise')
+
+const config = require('../config').getProperties()
+
+const TEMPLATES_WITH_NO_CODE = new Set([
+  'passwordResetEmail'
+])
 
 // SMTP half
 
-var MailParser = require('mailparser').MailParser
 var users = {}
 
 function emailName(emailAddress) {
   var utf8Address = Buffer.from(emailAddress, 'binary').toString('utf-8')
   return utf8Address.split('@')[0]
 }
 
-require('simplesmtp').createSimpleServer(
-  {
-    SMTPBanner: 'FXATEST'
-  },
-  function (req) {
-    var mp = new MailParser({ defaultCharset: 'utf-8' })
-    mp.on('end',
-      function (mail) {
-        var link = mail.headers['x-link']
-        var rc = mail.headers['x-recovery-code']
-        var rul = mail.headers['x-report-signin-link']
-        var uc = mail.headers['x-unblock-code']
-        var vc = mail.headers['x-verify-code']
-        var name = emailName(mail.headers.to)
-        if (vc) {
-          console.log('\x1B[32m', link, '\x1B[39m')
-        }
-        else if (rc) {
-          console.log('\x1B[34m', link, '\x1B[39m')
-        }
-        else if (uc) {
-          console.log('\x1B[36mUnblock code:', uc, '\x1B[39m')
-          console.log('\x1B[36mReport link:', rul, '\x1B[39m')
-
-        }
-        else {
-          console.error('\x1B[31mNo verify code match\x1B[39m')
-          console.error(mail)
-        }
-        if (users[name]) {
-          users[name].push(mail)
-        } else {
-          users[name] = [mail]
-        }
+module.exports = (printLogs) => {
+  printLogs = printLogs || process.env.MAIL_HELPER_LOGS
+  const console = printLogs ? global.console : {
+    log() {},
+    error() {}
+  }
+  return new P((resolve, reject) => {
+    const smtp = simplesmtp.createSimpleServer(
+      {
+        SMTPBanner: 'FXATEST'
+      },
+      function (req) {
+        var mp = new MailParser({ defaultCharset: 'utf-8' })
+        mp.on('end',
+          function (mail) {
+            var link = mail.headers['x-link']
+            var rc = mail.headers['x-recovery-code']
+            var rul = mail.headers['x-report-signin-link']
+            var uc = mail.headers['x-unblock-code']
+            var vc = mail.headers['x-verify-code']
+            var template = mail.headers['x-template-name']
+            var name = emailName(mail.headers.to)
+            if (vc) {
+              console.log('\x1B[32m', link, '\x1B[39m')
+            }
+            else if (rc) {
+              console.log('\x1B[34m', link, '\x1B[39m')
+            }
+            else if (uc) {
+              console.log('\x1B[36mUnblock code:', uc, '\x1B[39m')
+              console.log('\x1B[36mReport link:', rul, '\x1B[39m')
+            }
+            else if (TEMPLATES_WITH_NO_CODE.has(template)) {
```

Comment:
This was just because I kept seeing scary red `No verify code match` text output, even though some emails are expected to not have one.